File: 2.0/csharp/me/getCreditUsage.cs
using System;
using CallfireApiClient;
using CallfireApiClient.Api.Account.Model;
using CallfireApiClient.Api.Account.Model.Request;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var request = new DateIntervalRequest
        {
            IntervalBegin = new DateTime(2037, 12, 31, 20, 53, 20),
            IntervalEnd = new DateTime(2038, 1, 1, 8, 0, 0)
        };
        CreditsUsage creditsUsage = client.MeApi.GetCreditUsage(request);
    }
}

File: 2.0/java/me/getCreditUsage.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.account.model.CreditsUsage;
import com.callfire.api.client.api.account.model.request.DateIntervalRequest;

import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient callfireClient = new CallfireClient("api_login", "api_password");
        Date begin = new GregorianCalendar(2037, Calendar.DECEMBER, 31, 20, 53, 20).getTime();
        Date end = new GregorianCalendar(2038, Calendar.JANUARY, 1, 8, 0, 0).getTime();
        DateIntervalRequest request = DateIntervalRequest.create()
            .intervalBegin(begin)
            .intervalEnd(end)
            .build();
        CreditsUsage creditsUsage = callfireClient.meApi().getCreditUsage(request);
    }
}

